nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/impl/ACPImpl.java

```diff
@@ -46,6 +48,8 @@
 
     private transient Map<String, Access> cache;
 
+    private Boolean legacyBehavior;
```

Comment:
Why the boxed version?

Reply:
Because it acts like a cached value, we retrieve it only if it's null.